lib/zendesk_api/middleware/response/sanitise_response.rb

```diff
@@ -0,0 +1,13 @@
+module ZendeskAPI
+  module Middleware
+    module Response
+      class SanitiseResponse < Faraday::Response::Middleware
+        def on_complete(env)
+          unless env[:body].strip.empty?
+            env[:body] = env[:body].encode('UTF-8', 'binary', invalid: :replace, undef: :replace, replace: '')
```

Comment:
Nobody is on 2.0 ... right ? ... hardcode scrub ?

On Mon, Sep 7, 2015 at 9:13 AM, Gary Grossman <notifications@github.com>
wrote:

> In lib/zendesk_api/middleware/response/sanitise_response.rb
> <https://github.com/zendesk/zendesk_api_client_rb/pull/266#discussion_r38872008>
> :
>
> > @@ -0,0 +1,13 @@
> > +module ZendeskAPI
> > +  module Middleware
> > +    module Response
> > +      class SanitiseResponse < Faraday::Response::Middleware
> > +        def on_complete(env)
> > +          unless env[:body].strip.empty?
> > +            env[:body] = env[:body].encode('UTF-8', 'binary', invalid: :replace, undef: :replace, replace: '')
>
> Would it make sense to use String#scrub when it is available (Ruby 2.1
> and later)?
>
> We had similar issues for automatic credit card redaction and either use
> String#scrub or manually encode, see this code:
>
> https://github.com/zendesk/credit_card_sanitizer/blob/master/lib/credit_card_sanitizer.rb#L135-L147
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zendesk/zendesk_api_client_rb/pull/266/files#r38872008>
> .
>